modules/gapi/src/executor/gstreamingexecutor.cpp

```diff
@@ -521,6 +530,87 @@ void emitterActorThread(std::shared_ptr<cv::gimpl::GIslandEmitter> emitter,
     }
 }
 
+// This thread pulls data from the assigned input queues and makes sure that
+// all input args are in sync (timestamps are equal), dropping some inputs if required.
+// After getting synchronized inputs from all input queues, the thread pushes them to out queues
+void syncActorThread(std::vector<Q*> in_queues, std::vector<std::vector<Q*>> out_queues) {
```

Comment:
Vectors are being copied here

Reply:
Added moving of the parameters at the place this thread is created, there are specific rules of binding parameters to the thread storage, better to leave this signature as is